flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/OperationConverterUtils.java

```diff
@@ -117,6 +128,193 @@ public static Operation convertAddReplaceColumns(
                         catalogTable.getComment()));
     }
 
+    public static Operation convertAlterTableDropConstraint(
+            ObjectIdentifier tableIdentifier,
+            CatalogTable catalogTable,
+            SqlAlterTableDropConstraint alterTableDropConstraint) {
+        boolean isPrimaryKey = alterTableDropConstraint.isPrimaryKey();
+        Optional<Schema.UnresolvedPrimaryKey> oriPrimaryKey =
+                catalogTable.getUnresolvedSchema().getPrimaryKey();
+        // validate primary key is exists in table
+        if (!oriPrimaryKey.isPresent()) {
+            throw new ValidationException(
+                    String.format("Table %s does not exist primary key.", tableIdentifier));
+        }
+
+        String constraintName = null;
+        if (alterTableDropConstraint.getConstraintName().isPresent()) {
+            constraintName = alterTableDropConstraint.getConstraintName().get().getSimple();
+        }
+        if (!StringUtils.isNullOrWhitespaceOnly(constraintName)
+                && !oriPrimaryKey.get().getConstraintName().equals(constraintName)) {
+            throw new ValidationException(
+                    String.format(
+                            "CONSTRAINT [%s] does not exist in table %s",
+                            constraintName, tableIdentifier));
+        }
+
+        return new AlterTableDropConstraintOperation(tableIdentifier, isPrimaryKey, constraintName);
+    }
+
+    public static Operation convertDropWatermark(
+            ObjectIdentifier tableIdentifier, CatalogTable catalogTable) {
+        Schema originSchema = catalogTable.getUnresolvedSchema();
+        if (CollectionUtil.isNullOrEmpty(originSchema.getWatermarkSpecs())) {
+            throw new ValidationException(
+                    String.format("Table %s does not exist watermark.", tableIdentifier));
+        }
+
+        Schema.Builder builder = Schema.newBuilder();
+        // build column
+        builder.fromColumns(originSchema.getColumns());
+
+        // build primary key
+        Optional<Schema.UnresolvedPrimaryKey> originPrimaryKey = originSchema.getPrimaryKey();
```

Comment:
should we also build constraint or primary key is the only constraint?

Reply:
I did not understand the meaning of this sentence, can you give more context?

Reply:
Nerver mind. As the primary key is the only constraint, it's okay to only build primary key.